streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1229,10 +1229,21 @@ private void tryToLockAllNonEmptyTaskDirectories() {
             final String namedTopology = taskDir.namedTopology();
             try {
                 final TaskId id = parseTaskDirectoryName(dir.getName(), namedTopology);
-                if (stateDirectory.lock(id)) {
-                    lockedTaskDirectories.add(id);
-                    if (!allTasks.containsKey(id)) {
-                        log.debug("Temporarily locked unassigned task {} for the upcoming rebalance", id);
+                boolean lockedEmptyDirectory = false;
+                try {
+                    if (stateDirectory.lock(id)) {
+                        if (stateDirectory.directoryForTaskIsEmpty(id)) {
+                            lockedEmptyDirectory = true;
```

Comment:
This is maybe a bit pedantic, but I feel like it would be easier to understand the flow & intent of this code if we just unlocked it directly right here (instead of introducing the `lockedEmptyDirectory` flag and then unlocking it later on)

Also, since we log a debug message when we do lock the directory, we might as well put a similar debug message in this branch too. Something like `"Released lock on empty directory for task {}"`

Reply:
Thank you for the feedback! I updated to simplify the control flow and add a comment and log around the behavior. Please let me know how that looks.

Reply:
Hi @ableegoldman, can you take a look at the update please? Thank you